docs/reference/hooks.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: default
+title: Hooks
+parent: Reference
+nav_order: 10
+has_children: false
+---
+
+# Configurable Hooks
+Like other version control systems, lakeFS allows the configuration of `Actions` to trigger when predefined events occur.
+ 
+Supported Events:
+1. `pre_commit` - Action runs when the commit occurs, before the commit is finalized.
+2. `pre_merge` - Action runs when the merge occurs, before the merge is finalized.
+ 
+lakeFS `Actions` are handled per repository and cannot be reused.  
+Failure of any `Hook` under any `Action` will result in aborting the lakeFS operation that is taking place.
+
+## Action
+An `Action` is a list of `Hooks` with the same trigger configuration.
+`Hooks` list under an `Action` is ordered and an `Hook` will only be executed if all previous `Hooks` passed.  
+
+### Hook
+An `Hook` is the basic building block of an `Action`. 
+Failure of a single `Hook` will stop the execution of the containing `Action` and fail the `Run`. 
+
+### Action file
+Schema of the Action file:
```

Comment:
At this point in reading the doc, I don't know what an Action file is - perhaps explain the concept in a seperate paragraph prior to this one (hooks are managed by writing to prefix in the lakeFS repository, this allows configuration-as-code inside lakeFS, action files are declerative and written in YAML, ...). After such a paragraph, this one would make sense :)

Reply:
Done.